sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json

```diff
@@ -647,17 +648,17 @@
             "location": "[parameters('location')]"
         },
         {
-            "condition": "[and(variables('enablePE'), equals(parameters('storageAccountOption'), 'new'))]",
+            "condition": "[and(variables('enablePE'), or(equals(parameters('storageAccountOption'), 'new'), and(equals(parameters('set_up_feature_store'), 'true'), empty(parameters('offline_store_target')))))]",
             "type": "Microsoft.Storage/storageAccounts",
             "apiVersion": "2019-04-01",
-            "name": "[parameters('storageAccountName')]",
+            "name": "[variables('provisionedStorageAccountName')]",
```

Comment:
Is this variable needed? Is not it just the parameter('storageAccountName')

Reply:
this is needed, it represents the storage account name that needs to be provisioned, parameter('storageAccountName') could be user provided storage account name.